src/confetti.js

```diff
@@ -857,6 +857,39 @@
     };
   }
 
+  function shapesFromImage(imageData) {
+    var scalar = imageData.scalar != null ? imageData.scalar : 1;
+    var size = 10 * scalar;
+
+    var img = new Image();
+    img.src = imageData.src;
+
+    return promise(function (resolve) {
```

Comment:
Okay, so I know why you did this. It is so convenient for this lib to just load the image for you, even from a URL. I've seen this backfire a lot in libraries though, from folks who don't totally understand how image loading works thinking that the library has a bug when their images are not served as expected.

Not to mention, the image here doesn't include error handling (because why not add two problems to one comment, am I right?)

Anyway, I think a much more flexible API would be `shapeFromCanvas`. Have the user load anything they want into a canvas (could be an image, or you could draw anything in code using the canvas API), and let this lib just create the bitmap, scaling, other shape stuff.